migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -73,29 +79,31 @@ public PhoneRegion getRegionCode() {
 
   /**
    * Returns the sub range of numbers within numberRange that can be migrated using any recipe from
-   * the {@link CsvTable} recipesTable that matches the specified BCP-47 region code. This method will
-   * not perform migrations and as a result, the validity of migrations using the given recipesTable
-   * cannot be verified.
+   * the {@link CsvTable} recipesTable that matches the specified BCP-47 region code. This method
+   * will not perform migrations and as a result, the validity of migrations using the given
+   * recipesTable cannot be verified.
    */
-  public RangeTree getAllMigratableNumbers() {
+  public Stream<DigitSequence> getAllMigratableNumbers() {
     return RecipesTableSchema.toRangeTable(recipesTable)
-        .getRanges(RecipesTableSchema.REGION_CODE, regionCode)
-        .intersect(getNumberRange());
+        .getRanges(RecipesTableSchema.REGION_CODE, regionCode).asRangeSet()
+        .intersection(getNumberRange()).asRanges().stream().map(Range::lowerEndpoint);
   }
 
   /**
-   * Returns the sub range of numbers within numberRange that can be migrated using the given recipe.
-   * This method will not perform migrations and as a result, the validity of migrations using the
-   * given recipe cannot be verified.
+   * Returns the sub range of numbers within numberRange that can be migrated using the given
+   * recipe. This method will not perform migrations and as a result, the validity of migrations
+   * using the given recipe cannot be verified.
    *
    * @param recipeKey: the key of the recipe that is being checked
-   * @throws IllegalArgumentException if there is no row in the recipesTable with the given recipeKey
+   * @throws IllegalArgumentException if there is no row in the recipesTable with the given
+   * recipeKey
    */
-  public RangeTree getMigratableNumbers(RangeKey recipeKey) {
+  public Stream<DigitSequence> getMigratableNumbers(RangeKey recipeKey) {
```

Comment:
What use case did you have in mind for this method (i.e. when would the caller have a range key) ?

Reply:
Each recipe row is keyed by a RangKey. To check migratable numbers for a recipe (or migrate a range of numbers for a given recipe) the key is needed as it will represent all numbers that can match to a recipe